index.js

```diff
@@ -546,8 +549,15 @@ Command.prototype.executeSubCommand = function(argv, args, unknown) {
       proc = spawn(bin, args, { stdio: 'inherit', customFds: [0, 1, 2] });
     }
   } else {
-    args.unshift(bin);
-    proc = spawn(process.execPath, args, { stdio: 'inherit'});
+    if (isExplicitJS) {
+      args.unshift(bin);
+      // add executable arguments to spawn
+      args = (process.execArgv || []).concat(args);
+
+      proc = spawn('node', args, { stdio: 'inherit'});
```

Comment:
As for 'node', can you do it like https: //github.com/tj/commander.js/blob/master/index.js#L544?